feedhenry/fh-js-sdk
# FH-4398 - Dependency Security Vulnerabilities

## JIRA:
Link : https://issues.jboss.org/browse/FH-4398

## WHAT:
Fix High and Medium Security Vulnerabilities. 

## HOW:
* Upgrade the dependences browserify@3.46.1 and browserify-shim@3.3.2.
* Perform changes in the Gruntfile.js regards the breakchanges of this libs
* Add the dev dep `"phantomjs": "1.9.7-15"` to keep it compatible with NodeJS 4.
* Upgrade the dev dependece `grunt-browserify` for keep browserify grunt tasks running with success to test the project. 

### Changelogs of the dependencies upgraded:
* browserify : https://github.com/browserify/browserify/blob/master/changelog.markdown
* grunt-browserify : https://github.com/jmreidy/grunt-browserify/blob/master/CHANGELOG.md
* browserify-shim: https://gemnasium.com/npms/browserify-shim

## WHY
To solve the following vulnerabilities.
* Command Injection 
* Improper minification of non-boolean comparisons
* Insecure Randomness
* Regular Expression Denial of Service (DoS)
* Regular Expression Denial of Service (DoS)

## USAGE:

### Production:
To minify the files/code. It is used in the following task which is executed when the lib is installed (`npm install`). 

```
Running "browserify:dist" (browserify) task
found current version = 3.0.3
Version to inject is 3.0.3
>> Bundle dist/feedhenry.js created.
``` 

### Development:
Used to do the dist and in the grunt task executed to test the lib. (`$grunt`)

```
Running "browserify:dist" (browserify) task
found current version = 3.0.3
Version to inject is 3.0.3
>> Bundle dist/feedhenry.js created.

Running "browserify:require" (browserify) task
>> Bundle test/browser/feedhenry-latest-require.js created.

Running "browserify:test" (browserify) task
>> Bundle ./test/browser/browserified_tests.js created.
```

## Steps to test:
Check if the CI/Process finilized with succes, since it will install and run all grunt taks for the lib with the all supported NodeJS versions.
